### PR TITLE
Progress Indicator

### PR DESCRIPTION
**Pull request by rese**
## Overview

Added a Progress Indicator component that wraps the Fabric one. Progress Indicator is used to convey progress information of an ongoing operation such as a file upload. 

_YamUI ProgressIndicator:_
screen shot 2017-11-17 at 1 24 24 pm

_Fabric's ProgressIndicator:_
screen shot 2017-11-17 at 1 25 15 pm

## Pull request checklist

* [x] Component `README.md` file is up-to-date.
* [x] Component is unit tested.

### CODE REVIEW

**Comment by rese** [comment]
@unindented and @swese44 please review when you have the time. :)

File: src/components/ProgressIndicator/ProgressIndicator.css [hunk 1]
```diff
@@ -0,0 +1,10 @@
+.y-progress-indicator .ms-ProgressIndicator-progressTrack,
```
Comment (unindented) [inline src/components/ProgressIndicator/ProgressIndicator.css]: Missing copyright notice (sorry, we don't have that linting rule for CSS files)

File: src/components/ProgressIndicator/ProgressIndicator.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+### Notes for use
+
+`ProgressIndicator` spans the full width of its parent container. If you need to make it appear
+a certain width, you have to set the width property of its parent container to the desired
+width.
+
+If you need your ariaValueText translated, you'll have to do translation in the parent component and
+pass a translated string as `ProgressIndicator` will not be able to do it for you.
```
Comment (unindented) [inline src/components/ProgressIndicator/ProgressIndicator.md]: This applies to any label. @swese44 should we add this notice everywhere, or remove it from here?

Comment (swese44) [reply src/components/ProgressIndicator/ProgressIndicator.md]: ya i'd say probably just remove it

File: src/components/ProgressIndicator/ProgressIndicator.css (hunk 1 above)
Comment (rese) [reply src/components/ProgressIndicator/ProgressIndicator.css]: Added with: https://github.com/Microsoft/YamUI/pull/122/commits/3226384ac81882e0c72f85e9a44ce751acc62f94

File: src/components/ProgressIndicator/ProgressIndicator.md (hunk 2 above)
Comment (rese) [reply src/components/ProgressIndicator/ProgressIndicator.md]: Removed with: https://github.com/Microsoft/YamUI/pull/122/commits/ce18ac01a4b37fb2048e11575b1481a50e49c523

File: src/components/ProgressIndicator/ProgressIndicator.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+<ProgressIndicator percentComplete={0} ariaValueText="Zero percent" />
+```
+
+Thirty Percent:
+
+```js { "props": { "data-example": "thirty percent" } }
+<ProgressIndicator percentComplete={0.3} ariaValueText="Thirty percent" />
+```
+
+One Hundred Percent:
```
Comment (unindented) [inline src/components/ProgressIndicator/ProgressIndicator.md]: Why are these capitalized?

Comment (rese) [reply src/components/ProgressIndicator/ProgressIndicator.md]: oopps. 😨

Comment (rese) [reply src/components/ProgressIndicator/ProgressIndicator.md]: Uncapitalized with: https://github.com/Microsoft/YamUI/pull/122/commits/5e85ff8e9d013e536a494bbe2d68fe3c474f9be5

**Comment by rese** [comment]
Thanks @mamuso!

**Comment by rese** [comment]
@swese44 I scoped the color overrides under our y- class: https://github.com/Microsoft/YamUI/pull/122/commits/6850ce91ba871acba8e7f915c48518dc7ffb147c

**Comment by swese44** [comment]
i just added one more tiny change to use semantic color variables instead of `indieRock` and `river`

**Comment by mamuso** [comment]
@swese44 does it make sense in this context making the color variable this abstract?

**Comment by rese** [comment]
Are the variable names $borderColor and $linkColor descriptive enough for our purposes?

**Comment by rese** [comment]
Got it. I'm ok with the changes.

**Comment by swese44** [comment]
ok let's merge so we can publish to npm, but @mamuso we can create new variables any time if progressIndicatorBGColor shouldn't be tied to defaultBorderColor, or progressIndicatorBar shouldn't be tied to linkColor

**Comment by mamuso** [comment]
Yeah, we will talk about it. My concern is that linkColor conveys a pretty narrow case, and six months from now we might update it without understanding the impact that this could have on components like a progress indicator.

**Comment by swese44** [comment]
sounds good. at least we have visual diffs to prevent them from breaking in the meantime

**Pull request merged**